src/backend/routers/student.test.ts

```diff
@@ -81,6 +81,56 @@ test("addIep and getIep", async (t) => {
   t.deepEqual(got[0].end_date, added.end_date);
 });
 
+test("editIep", async (t) => {
+  const { trpc, seed } = await getTestServer(t, {
+    authenticateAs: "case_manager",
+  });
+
+  // * must add student this way to populate the assigned_case_manager_id
+  await trpc.case_manager.addStudent.mutate({
+    first_name: seed.student.first_name,
+    last_name: seed.student.last_name,
+    email: seed.student.email,
+    grade: seed.student.grade,
+  });
+
+  const myStudents = await trpc.case_manager.getMyStudents.query();
+  const student_id = myStudents[0].student_id;
+
+  const iep = await trpc.student.addIep.mutate({
+    student_id: student_id,
+    start_date: new Date("2023-01-01"),
+    end_date: new Date("2023-12-31"),
+  });
+
+  let got = await trpc.student.getIeps.query({
+    student_id: student_id,
+  });
+
+  t.is(got.length, 1);
+  t.is(got[0].student_id, iep.student_id);
+  t.deepEqual(got[0].start_date, iep.start_date);
+  t.deepEqual(got[0].end_date, iep.end_date);
+
+  const updated_start_date = new Date("2023-03-02");
+  const updated_end_date = new Date("2024-03-01");
+
+  await trpc.student.editIep.mutate({
+    student_id: student_id,
+    start_date: updated_start_date,
+    end_date: updated_end_date,
+  });
+
+  got = await trpc.student.getIeps.query({
+    student_id: student_id,
+  });
+
+  t.is(got.length, 1);
+  t.is(got[0].student_id, student_id);
+  t.notDeepEqual(got[0].start_date, iep.start_date);
```

Comment:
I don't think this part of the test is really capturing what you want since it's just checking to see if the date is changed, especially if there's still weird date stuff going on.